Upload Inventory By Vendor

aditiyars opened the pull request:
Create Unit test
add feature Create Inventory By Vendor on inventory controller
add findyById on seller repository
Create UploadInventoryRequest

close #36

aditiyars committed:
Create unit test
add feature Create Inventory By Vendor on inventory controller
add findyById on seller repository
Create UploadInventoryRequest

faisalmorensya commented on src/main/java/com/unsri/ecommerce/infrastructure/repository/SellerRepository.java:
```diff
@@ -9,4 +9,5 @@
 @Repository
 public interface SellerRepository extends JpaRepository<Seller, Integer> {
     Optional<Seller> findByEmail(String email);
+    Optional<Seller> findById(Integer id);
```
ini dipakai untuk apa?

faisalmorensya commented on src/main/java/com/unsri/ecommerce/presentation/controllers/InventoryController.java:
```diff
@@ -26,12 +28,14 @@
 public class InventoryController extends BaseController {
 
     private InventoryRepository inventoryRepository;
+    private SellerRepository sellerRepository;
```
repository tidak digunakan dimanapun?
jika tidak dipakai, maka seharusnya dihapus

aditiyars replied:
sudah dihapus

faisalmorensya commented on src/main/java/com/unsri/ecommerce/presentation/controllers/InventoryController.java:
```diff
@@ -87,9 +91,17 @@ public InventoryController(InventoryRepository inventoryRepository) {
+
         CreateInventory command = new CreateInventory(inventoryRepository);
-        return command.execute(Optional.ofNullable(item));
+        command.execute(Optional.of(inventory));
```
buat pengecekan jika proses sudah berhasil atau tidak

faisalmorensya commented on src/main/java/com/unsri/ecommerce/presentation/controllers/InventoryController.java:
```diff
@@ -87,9 +91,17 @@ public InventoryController(InventoryRepository inventoryRepository) {
+        command.execute(Optional.of(inventory));
+        BaseResponse baseResponse= new BaseResponse<>();
+
+        baseResponse.setMessage("Create Inventory Successfully");
```
pesan sukses dibuat berdasarkan pengecekan jika proses sudah berhasil

faisalmorensya commented on src/test/java/com/unsri/ecommerce/application/behaviours/inventory/commands/CreateInventoryTests.java:
```diff
@@ -26,11 +27,22 @@ public CreateInventoryTests() {
-        inventory.setPrice(1000.0);
+        inventory.setItemName(inventoryRequest.getProductName());
+        inventory.setPrice(inventoryRequest.getPrice());
+        inventory.setPhotos(inventoryRequest.getPhoto());
```
foto berisi data kosong, masukkan data seperti real world

aditiyars replied:
done

faisalmorensya reviewed (commented):
please check request perubahan dan check conflict

faisalmorensya commented:
tolong isi 
- bagian Linked Issues
- bagian Projects

aditiyars committed:
Create unit test
add feature Create Inventory By Vendor on inventory controller
Create UploadInventoryRequest
add cascade on Inventory for fk_inventory_id
add annotation Column on PhotoInventory for fk_inventory_id
add javax.validation depedency on pom.xml
delete findById on SellerRepository
add annotation @NotEmpty and @Positive on UploadInventoryRequest for productName and price

aditiyars committed:
Create Product By Vendor

Create unit test
add feature Create Inventory By Vendor on inventory controller
Create UploadInventoryRequest
add cascade on Inventory for fk_inventory_id
add annotation Column on PhotoInventory for fk_inventory_id
add javax.validation depedency on pom.xml
delete findById on SellerRepository
add annotation @NotEmpty and @Positive on UploadInventoryRequest for productName and price
merge from dev

aditiyars committed:
Create Product By Vendor

add depedency spring validation

faisalmorensya commented on src/main/java/com/unsri/ecommerce/domain/models/Inventory.java:
```diff
@@ -21,7 +24,7 @@
     private Integer fkSellerId;
 
-    @OneToMany(targetEntity = PhotoInventory.class)
+    @OneToMany(targetEntity = PhotoInventory.class, cascade = CascadeType.ALL)
```
CascadeType.ALL digunakan untuk apa?

aditiyars replied:
Cascade type all ini sebenernya bukan bagian dari fitur create inventory tapi cascade type all ini digunakan untuk menghapus/update seluruh data dengan id yang kita pilih

faisalmorensya replied:
bisa kasih contoh? @aditiyars

aditiyars replied:
contohnya pada satu seller bisa memiliki banyak(lebih dari satu) foto untuk satu seller, kalo gak pake cascade bakal munculin transient exception (TransientObjectException occurs when you save an object which references another object that is transient (meaning it has the "default" identifier value, frequently null) and then flush the Session.)

faisalmorensya commented:
closing and get the main implementation for refactoring purpose
